test/suite/helper_tiered.py

```diff
@@ -260,22 +260,44 @@ def tiered_conn_extensions(self, extlist):
         extlist.extension('storage_sources', self.ss_name + config)
 
     def download_objects(self, bucket_name, prefix):
-        import boto3
-        # The bucket from the storage source is expected to be a name and a region, separated by a 
-        # semi-colon. eg: 'abcd;ap-southeast-2'.
-        bucket_name, region = bucket_name.split(';')
-        
-        # Get the bucket resource and list the objects within that bucket that match the prefix for a
-        # given test.
-        s3 = boto3.resource('s3')
-        bucket = s3.Bucket(bucket_name)
-        objects = list(bucket.objects.filter(Prefix=prefix))
-
         # Create a directory within the test directory to download the objects to.
-        s3_object_files_path = 's3_objects/'
-        if not os.path.exists(s3_object_files_path):
-            os.makedirs(s3_object_files_path)
+        object_files_path = 'objects/'
+        if not os.path.exists(object_files_path):
+            os.makedirs(object_files_path)
+
+        if (self.ss_name == 's3_store'):
+            import boto3
+            # The bucket from the storage source is expected to be a name and a region, separated by a 
+            # semi-colon. eg: 'abcd;ap-southeast-2'.
+            bucket_name, region = bucket_name.split(';')
+            
+            # Get the bucket resource and list the objects within that bucket that match the prefix for a
+            # given test.
+            s3 = boto3.resource('s3')
+            bucket = s3.Bucket(bucket_name)
+            objects = list(bucket.objects.filter(Prefix=prefix))
+
+            for o in objects:
+                file_path = object_files_path + '/' + o.key.split('/')[-1]
+                bucket.download_file(o.key, file_path)
+        elif (self.ss_name == 'gcp_store'):
+            from google.cloud import storage
+            
+            storage_client = storage.Client()
+            bucket = storage_client.bucket(bucket_name)
+            blobs = storage_client.list_blobs(bucket_name, prefix=prefix)
+
+            for blob in blobs:
+                file_path = object_files_path + '/' + blob.split('/')[-1]
+                blob.download_to_filename(file_path)
+        elif (self.ss_name == 'azure_store'):
+            from azure.storage.blob import BlobServiceClient
+
+            blob_service_client = BlobServiceClient.from_connection_string(self.auth_token) 
+            container_client = blob_service_client.get_container_client(container=bucket_name) 
+            blob_list = container_client.list_blobs()
 
-        for o in objects:
-            filename = s3_object_files_path + '/' + o.key.split('/')[-1]
-            bucket.download_file(o.key, filename)
+            for blob in blob_list:
+                file_path = object_files_path + '/' + blob.name.split('/')[-1]
+                with open(file=file_path, mode="wb") as download_file:
+                    download_file.write(container_client.download_blob(blob.name).readall())
```

Comment:
I think we should have a catch-all `else` here that throws an exception, in case someone adds another storage service, changes the formatting of `ss_name`, etc.

Reply:
yep I have now modified it to be more error prone if things go wrong let me know what you think [22b35ac](https://github.com/wiredtiger/wiredtiger/pull/8953/commits/22b35acd1575bfed2e228baece7e3d1775026f83)